metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/asa/BasicAsaParser.java

```diff
@@ -0,0 +1,165 @@
+		    .put("ASA-7-710002", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710003", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710005", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710006", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-6-713172", "CISCOFW713172")
+		    .put("ASA-4-733100", "CISCOFW733100")
+		    .put("ASA-6-305012", "CISCOFW305012")
+		    .put("ASA-7-609001", "CISCOFW609001")
+		    .put("ASA-7-609002", "CISCOFW609002")
+            .put("ASA-5-713041", "CISCOFW713041")
+            .build();
+
+    @Override
+    public void configure(Map<String, Object> config) {
+
+    }
+
+    @Override
+    public void init() {
+        asaGrok = new Grok();
+        InputStream patternStream = this.getClass().getClassLoader().getResourceAsStream("patterns/asa");
+        try {
+            asaGrok.addPatternFromReader(new InputStreamReader(patternStream));
+        } catch (GrokException e) {
+            e.printStackTrace();
+        }
+        LOG.info("[Metron] CISCO ASA Parser Initialized");
+    }
+
+    @Override
+    public List<JSONObject> parse(byte[] rawMessage) {
+        String syslogPattern = "%{CISCO_TAGGED_SYSLOG}";
+        JSONObject metronJson = new JSONObject();
+        List<JSONObject> messages = new ArrayList<>();
+        try {
+            String logLine = new String(rawMessage, "UTF-8");
+            LOG.trace("[Metron] Started parsing raw message: " + logLine);
+
+            asaGrok.compile(syslogPattern);
+            Match syslogMatch = asaGrok.match(logLine);
+            syslogMatch.captures();
+            Map<String, Object> syslogJson = syslogMatch.toMap();
+            LOG.trace("[Metron] Grok CISCO syslog matches: " + syslogMatch.toJson());
+
+            metronJson.put(Constants.Fields.ORIGINAL.getName(), logLine);
+            metronJson.put(Constants.Fields.TIMESTAMP.getName(),
+                    SyslogUtils.parseTimestampToEpochMillis((String) syslogJson.get("CISCOTIMESTAMP")));
+            metronJson.put("ciscotag", syslogJson.get("CISCOTAG"));
+            metronJson.put("syslog_severity", SyslogUtils.getSeverityFromPriority((int) syslogJson.get("syslog_pri")));
+            metronJson.put("syslog_facility", SyslogUtils.getFacilityFromPriority((int) syslogJson.get("syslog_pri")));
+
+            String messagePattern = patternMap.get(syslogJson.get("CISCOTAG"));
+            asaGrok.compile("%{"+ messagePattern +"}");
+            Match messageMatch = asaGrok.match((String) syslogJson.get("message"));
+            messageMatch.captures();
+            Map<String, Object> messageJson = messageMatch.toMap();
+            LOG.trace("[Metron] Grok CISCO message matches: " + messageMatch.toJson());
+
+            String src_ip = (String) messageJson.get("src_ip");
+            if (src_ip != null && ipValidator.isValid(src_ip))
+                metronJson.put(Constants.Fields.SRC_ADDR.getName(), src_ip);
```

Comment:
It turns out the `ipValidator` function isn't adding any benefit here. The grok pattern being used on the raw message is already checking that it's a valid IP address (IPv4 or IPv6). Given that I'm going to simply remove that validation from the code as redundant.